server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java

```diff
@@ -1252,8 +1253,7 @@ public Analysis visitInsertRow(InsertRowStatement insertRowStatement, MPPQueryCo
     dataPartitionQueryParam.setTimePartitionSlotList(insertRowStatement.getTimePartitionSlots());
 
     DataPartition dataPartition =
-        partitionFetcher.getOrCreateDataPartition(
-            Collections.singletonList(dataPartitionQueryParam));
+        getDataPartition(Collections.singletonList(dataPartitionQueryParam));
 
     Analysis analysis = new Analysis();
```

Comment:
Check whether the taken dataPartition is null, and end the task directly if null.

Reply:
After my check, DataPartition may be empty, but not null.